backend/packages/Upgrade/src/api/services/AnalyticsService.ts

```diff
@@ -188,7 +193,51 @@ export class AnalyticsService {
         if (experiment.assignmentUnit === ASSIGNMENT_UNIT.WITHIN_SUBJECTS) {
           csvExportData = await this.analyticsRepository.getCSVDataForWithInSubExport(experimentId, skip, take);
         } else {
-          csvExportData = await this.analyticsRepository.getCSVDataForSimpleExport(experimentId, skip, take);
+          // Get the experiment details
+          const experimentRepository = getCustomRepository(ExperimentRepository, 'export');
+          const experimentQuery = experimentRepository
```

Comment:
This is a copy of the function in `ExperimentService` -> `findOne`. Can you refactor this code and move it to the Experiment Repository?

Reply:
This is subset of the `findOne`. In `findOne` function we have a lot of extra joins and selections which we dont require here. If you say, I can move this query in a separate function `findExperimentForCSVExport` in the `ExperimentService ` and call it from there.

Reply:
Yup move it to the repository and reuse the common code.